### ISSUE TITLE
Route.push not works with full url

### ISSUE THREAD

**Issue by BananaWanted** [isaachinman/next-i18next#245]
## Describe the bug
```javascript
Router.push('http://my.domain/en-US/')
```
ends up to `http://my.domain/en-US/en-UShttp://my.domain/en-US/` in the browser

## Occurs in next-i18next version
0.36.5 (current version as of Mar 22 2019)

## Steps to reproduce
Simply write `Router.push('http://my.domain/en-US/')` in your app

## Expected behavior
`Router.push('http://my.domain/en-US/')` should bring me to `http://my.domain/en-US/`

## Screenshots
N/A

## Additional context
I looked into the code a bit
And [found in this line](https://github.com/isaachinman/next-i18next/blob/master/src/utils/lng-path-corrector.js#L58):
```
    as = `/${currentLanguage}${as}`
```
`next-i18next` is arbitrarily prepend `/${currentLanguage}` to the target url.

**Comment by isaachinman** [comment]
Does that work in NextJs? What's the reasoning for non-relative paths?

**Comment by BananaWanted** [comment]
It works with the nextjs router. 

The use case here is we need a page as a unified place to handle redirections, for example, OAuth callbacks. The URLs passed to this page could be in any form (`http://my.domain/path`, `/en-US/path`, `/path`) as long as it points to some page on the site.

**Comment by isaachinman** [comment]
> It works with the nextjs router

Okay, then we need to fix this.

@capellini Any interest?

**Comment by capellini** [comment]
@isaachinman Sure, I'll write up a PR in the next few days.

**Comment by isaachinman** [comment]
Fixed with 2aa2204, released with v0.37.0
